lib/logger.c

```diff
@@ -321,7 +326,7 @@ grn_logger_output_end(grn_ctx *ctx,
     }
   }
 #ifndef _WIN32
-  if (output->need_flock) {
+  if (output->need_flock && output->file) {
```

Comment:
Could we call `flock(fileno(output->file), LOCK_UN);` like the following?
Because I'm no sure whether we need to call it when `written` is <= 0 or not mm.

```c
    if (output->file != stdout &&
        output->file != stderr &&
        (output->rotate_threshold_size > 0 &&
         output->size >= output->rotate_threshold_size)) {
#ifndef _WIN32
      if (output->need_flock) {
        flock(fileno(output->file), LOCK_UN);
      }
#endif
      fclose(output->file);
      output->file = NULL;
      grn_logger_output_rotate(ctx, output);
    } else {
      fflush(output->file);
      if (output->need_flock) {
        flock(fileno(output->file), LOCK_UN);
      end
    }
```

Reply:
We need `flock(LOCK_UN)` even when `written == 0`.

Reply:
`flock(LOCK_UN)` must be balanced with `flock(LOCK_EX)`.

Reply:
Thank you very much. I understand that we need to unlock it even if the write operation results in 0, because the file has already been locked.